src/librustdoc/clean/inline.rs

```diff
@@ -324,17 +324,19 @@ fn merge_attrs(
     // doc comments show up before the original doc comments
     // when we render them.
     if let Some(inner) = new_attrs {
-        if let Some(new_id) = parent_module {
-            let diag = cx.sess().diagnostic();
-            Attributes::from_ast(diag, old_attrs, Some((inner, new_id)))
-        } else {
-            let mut both = inner.to_vec();
-            both.extend_from_slice(old_attrs);
-            both.clean(cx)
+        if !inner.is_empty() {
```

Comment:
`inner` includes all attributes on the `pub use` statement not just doc comments. This means the `std` docs aren't fixed by this because all `pub use` statements have at least a stability attribute.